src/expr/src/expr/expr_regexp.rs

```diff
@@ -248,3 +252,513 @@ impl Expression for RegexpMatchExpression {
         })
     }
 }
+
+#[derive(Debug)]
+pub struct RegexpReplaceExpression {
+    /// The source to be matched and replaced
+    pub source: Box<dyn Expression>,
+    /// The regex context, used to match the given pattern
+    pub ctx: RegexpContext,
+    /// The replacement string
+    pub replacement: String,
+    /// The actual return type by evaluating this expression
+    pub return_type: DataType,
+    /// The start position to replace the source
+    /// The starting index should be `0`
+    pub start: i32,
+    /// The N, used to specified the N-th position to be replaced
+    /// Note that this field is only available if `start` > 0
+    pub n: i32,
+    /// Indicates if the `-g` flag is specified
+    pub global_flag: bool,
```

Comment:
I think `global_flag` should be put into `RegexpOptions` for consistency. Then we can change `RegexpContext` to:
```rust
pub struct RegexpContext {
    regex: Regex,
    global: bool,
}
```

Btw, should we also enable the [unicode mode](https://docs.rs/regex/latest/regex/struct.RegexBuilder.html#method.unicode) for the regex? 🤔️

Reply:
Ahh I see, then it would be more natural to directly use `ctx.global` instead of a separate `global_flag`.
Now the source text with unicode works well (in `e2e_test`) , but I do not know if the unicode mode for `regex` has been enabled or not.